lib/response.js

```diff
@@ -766,13 +766,31 @@ internals.Response.Peek = class extends Stream.Transform {
         super();
 
         this._podium = podium;
-        this.on('finish', () => podium.emit('finish'));
     }
 
-    _transform(chunk, encoding, callback) {
+    async _transform(chunk, encoding, callback) {
+
+        try {
+            await this._podium.emit('peek', [chunk, encoding]);
+        }
+        catch (err) {
+            callback(err);
+            return;
+        }
+
+        callback(null, chunk);
+    }
+
+    async _final(callback) {
+
+        try {
+            await this._podium.emit('finish');
+        }
+        catch (err) {
+            callback(err);
```

Comment:
no sure why you'd want to call `this.destroy(err)` directly. It [seems](https://github.com/nodejs/node/blob/9c62e0e384e2ea5a45a4011f31c97be60f2a0737/lib/internal/streams/writable.js#L651-L657) that that's what happening under the hood anyway (respecting the `autoDestroy` option).